SharedContent/cs-winui-template/WinUI3TemplateCs/MainPage.xaml.cs

```diff
@@ -0,0 +1,156 @@
+            // here to load the home page.
+            if (scenarios is not null && scenarios.Count > 0)
+            {
+                NavView_Navigate(scenarios.First().ClassName, new Microsoft.UI.Xaml.Media.Animation.EntranceNavigationTransitionInfo());
+            }
+        }
+
+        private void NavView_Navigate(string navItemTag, Microsoft.UI.Xaml.Media.Animation.NavigationTransitionInfo transitionInfo)
+        {
+            Type _page = null;
+
+            if (navItemTag == nameof(SettingsPage))
+            {
+                _page = typeof(SettingsPage);
+            }
+            else
+            {
+                var item = scenarios.FirstOrDefault(p => p.ClassName.Equals(navItemTag));
+                _page = Type.GetType(item.ClassName);
+            }
+
+            // Get the page type before navigation so you can prevent duplicate
+            // entries in the backstack.
+            var preNavPageType = ContentFrame.CurrentSourcePageType;
+
+            // Only navigate if the selected page isn't currently loaded.
+            if (!(_page is null) && !Type.Equals(preNavPageType, _page))
+            {
+                ContentFrame.Navigate(_page, null, transitionInfo);
+            }
+        }
+
+        private void NavView_BackRequested(NavigationView sender, NavigationViewBackRequestedEventArgs args)
+        {
+            if (ContentFrame.CanGoBack)
+            {
+                ContentFrame.GoBack();
+            }
+        }
+
+        private void NavView_ItemInvoked(NavigationView sender, NavigationViewItemInvokedEventArgs args)
+        {
+            var naViewItemInvoked = (NavigationViewItem)args.InvokedItemContainer;
+
+            if (args.IsSettingsInvoked == true)
+            {
+                NavView_Navigate(nameof(SettingsPage), args.RecommendedNavigationTransitionInfo);
+            }
+            else
+            if (args.InvokedItemContainer is not null)
+            {
+                var navItemTag = args.InvokedItemContainer.Tag?.ToString();
+                if (navItemTag is not null)
+                {
+                    NavView_Navigate(navItemTag, new Microsoft.UI.Xaml.Media.Animation.EntranceNavigationTransitionInfo());
+                }
+            }
+        }
+
+        private void ContentFrame_Navigated(object sender, NavigationEventArgs e)
+        {
+            NavView.IsBackEnabled = ContentFrame.CanGoBack;
+
+            if (ContentFrame.SourcePageType == typeof(SettingsPage))
+            {
+                // SettingsItem is not part of NavView.MenuItems, and doesn't have a Tag.
+                NavView.SelectedItem = (NavigationViewItem)NavView.SettingsItem;
+                NavView.Header = "Settings";
+            }
+            else
+            if (ContentFrame.SourcePageType != null)
+            {
+                var item = scenarios.FirstOrDefault(p => p.ClassName == e.SourcePageType.FullName);
```

Comment:
Another use of `FirstOrDefault` followed by dereferencing of the result without a null check.

Reply:
If you are going to assume that the result is never null, then use `First()` rather than `FirstOrDefault()`. `First()` requires that a match be found. `FirstOrDefault()` says "I'll look for one, but if I can't find one, I'll return null." Don't ask for a feature that makes you crash. Let `First` report the error. That makes the problem much more diagnosable.